.circleci/config.yml

```diff
@@ -0,0 +1,101 @@
+version: 2
+
+buildSteps: &buildSteps
+  - run:
+      name: Ensure npm is at least at version 3
+      command: |
+        if [[ "$(npm --version)" = [12].* ]]; then
+          sudo npm install --global npm@3
+        fi
+  - checkout
+  - run:
+      name: Install
+      command: npm install --loglevel warn
+  - run:
+      name: Test
+      command: npm test --loglevel warn
+  - persist_to_workspace:
+      # Persist all job output, so we can (potentially) use it for deploys
+      root: ../
+      paths:
+        - ./node-*
+
+jobs:
+  node-4:
+    docker:
+      - image: circleci/node:4
+    working_directory: ~/node-4
+    steps: *buildSteps
```

Comment:
Arguably we don't need 4 any more - as of now it's out of LTS, and there's quite a few places where we already don't support it.

Reply:
Yeah that's true, but this version was designed to support node 4 and has dependents that expect it, and it's simple to have the.  That said I'd happily have a major breaking version that removes that support if there's ever any need for it (the fact we hadn't even needed to add tests/auto-publishing before kind of shows how often we need to update it :P )